module/api/action/character/character-use-item-action.js

```diff
@@ -13,12 +13,49 @@ export const characterUseItemAction = async (actor, item, outcome = null, chatMe
     return;
   }
 
-  const [compendium, macroName] = compendiumInfoFromString(item.actionMacro);
-  if (compendium && macroName) {
-    const macro = await findCompendiumItem(compendium, macroName);
-    await executeMacro(macro, { actor, item, outcome, chatMessage });
-  } else {
-    const macro = game.macros.find((m) => m.name === macroName);
-    await executeMacro(macro, { actor, item, outcome, chatMessage });
+  const actionValue = String(item.actionMacro).trim();
+
+  /** @type {Document|undefined} */
+  let target;
+
+  // 1) UUID support (world or compendium): Macro.<id> / RollTable.<id> or Compendium.<pack>.<id>
+  if (actionValue.startsWith("Macro.") || actionValue.startsWith("RollTable.") || actionValue.startsWith("Compendium.")) {
+    try {
+      target = await fromUuid(actionValue);
+    } catch (e) {
+      // ignore and fall through to other strategies
+    }
+  }
+
+  // 2) Compendium + name format: "compendium;name"
+  if (!target && actionValue.includes(";")) {
+    const [compendium, name] = compendiumInfoFromString(actionValue);
+    if (compendium && name) {
+      target = await findCompendiumItem(compendium, name);
+    }
   }
+
+  // 3) World macro or rolltable by name
+  if (!target) {
+    const name = actionValue;
+    target = game.macros.find((m) => m.name === name) || game.tables?.find?.((t) => t.name === name);
+  }
+
+  if (!target) {
+    ui.notifications?.warn?.(`Action target not found: ${actionValue}`);
+    return;
+  }
+
+  // Execute Macro or draw RollTable
+  const documentName = /** @type {any} */ (target).documentName || /** @type {any} */ (target).entity; // compatibility
+  if (documentName === "Macro") {
+    await executeMacro(target, { actor, item, outcome, chatMessage });
+    return;
```

Comment:
```suggestion
```

Github doesn't seem to want to let me remove this line as part of the above suggestion.